devel/fortio/http.go

```diff
@@ -118,7 +132,32 @@ func FetchURL(url string) (int, []byte) {
 	}
 	code := resp.StatusCode
 	if Verbosity > 1 {
-		log.Printf("Got %d : %s for %s - response is %d bytes", code, resp.Status, url, len(data))
+		log.Printf("Got %d : %s for %s - response is %d bytes", code, resp.Status, c.url, len(data))
 	}
 	return code, data
 }
+
+// NewClient creates a client object
+func NewClient(url string, numConnections int, compression bool) *Client {
+	req := newHTTPRequest(url)
+	if req == nil {
+		return nil
+	}
+	client := Client{
+		url,
+		req,
+		&http.Client{
+			Timeout: 3 * time.Second,
```

Comment:
it means (I think) the max duration will be 3000ms before giving up - given we're expecting data in the single digit ms range most of the time, it seemed enough but I don't really have a good reason - probably should make it configurable or not set it